tests/integrationv2/test_buffered_send.py

```diff
@@ -0,0 +1,161 @@
+import pytest
+
+from configuration import available_ports, PROTOCOLS, ALL_TEST_CIPHERS, Certificates
+from common import ProviderOptions, data_bytes
+from fixtures import managed_process # lgtm [py/unused-import]
+from providers import Provider, S2N, OpenSSL, GnuTLS
+from utils import invalid_test_parameters, get_parameter_name, to_bytes, is_subsequence
+
+SEND_DATA_SIZE = 2 ** 14
+
+K_BYTES = 1024 
+SEND_BUFFER_SIZE_MIN = 1031
```

Comment:
I don't think it's overkill. The minimum send buffer size could trigger fragmentation of alert messages witch is an indication that we allow alerts to fragment (which is something to avoid -- re:GNU TLS). 

We should have a way to ensure we don't do that.

Reply:
-- I'll work on this.

Reply:
I added a sanity check that should fail if the minimum ever changes: `test_SEND_BUFFER_SIZE_MIN_is_s2ns_min_buffer_size`.